cli/pg.go

```diff
@@ -160,11 +166,28 @@ func runPgRestore(args *docopt.Args, client controller.Client, config *runConfig
 		defer bar.Finish()
 		config.Stdin = bar.NewProxyReader(config.Stdin)
 	}
-	return pgRestore(client, config)
+
+	return pgRestore(client, config, jobs)
 }
 
-func pgRestore(client controller.Client, config *runConfig) error {
-	config.Args = []string{"pg_restore", "-d", config.Env["PGDATABASE"], "--clean", "--if-exists", "--no-owner", "--no-acl"}
+func pgRestore(client controller.Client, config *runConfig, jobs int) error {
+	// Provision a volume at /data to stream the dump to.
+	if jobs > 1 {
+		// Check if controller supports data volumes for one-off jobs
+		compatMessage := "Cluster versions prior to %s don't support the --jobs argument for parallel restore."
+		if err := compatCheck(client, "v20161018.0", compatMessage); err != nil {
```

Comment:
This is actually a dependency on the controller version (I added support for data volumes for one off jobs in this PR), but yeah, it would be nice if the Postgres version was also exposed in the status details.

The GraphQL stuff sounds good, if the error returned indicated which field was the problem that would probably solve most of the backwards compatible error handling.